src/utilities/identities/identities.ts

```diff
@@ -378,11 +384,14 @@ async function syncDidStateWithBlockchain(address: string | null | undefined) {
     return;
   }
 
-  const { lightDid, fullDid, type } = parseDidUrl(identity.did);
+  const { lightDid, fullDid, type } = parseDidUri(identity.did);
+  const wasOnChain = type === 'full';
 
-  const isOnChain = Boolean(await DidChain.queryDidDetails(fullDid));
+  const resolved = await DidResolver.resolveDoc(identity.did);
+  const isOnChain = wasOnChain
+    ? Boolean(resolved && resolved.metadata && !resolved.metadata.deactivated)
+    : Boolean(resolved && resolved.metadata && resolved.metadata.canonicalId);
```

Comment:
@Diiaablo95 that check also had to be fixed and became rather inconvenient

Reply:
When compared with 
```js 
const isOnChain = Boolean(await DidChain.queryDidDetails(fullDid));

Reply:
Sounds good!

Reply:
Is it good that it is inconvenient? ;)

Reply:
Sorry I misread the message! What is the goal of this set of expressions? To check whether a stored DID is a full DID, and if so if it is stored on chain. Right?

Reply:
Yes, this is how we synchronize the stored DID with the blockchain. If we have a light DID stored, could be that it was upgraded elsewhere, so now we need to store the full DID. If we have a full DID stored, could be that is was downgraded elsewhere, in that case we need to store the light DID.